### PR TITLE
Last php cs fixer php migration rules

### PR DESCRIPTION
**Pull request by jack-worman**
* Updated my author stuff
* Added PHP 7.4 migration rules
* Updated composer.json homepage
* Added PHP 8.1 migration rules

### CODE REVIEW

File: phpseclib/Crypt/EC/Formats/Keys/PKCS8.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public static function load($key, ?string $password = null): array
             throw new \UnexpectedValueException('Key should be a string - not a ' . gettype($key));
         }
 
-        $isPublic = strpos($key, 'PUBLIC') !== false;
+        $isPublic = str_contains($key, 'PUBLIC')  ;
```
Comment (terrafrost) [inline phpseclib/Crypt/EC/Formats/Keys/PKCS8.php:82]: Trailing white space should be removed here

File: phpseclib/Crypt/RSA/Formats/Keys/PSS.php [hunk 2]
```diff
@@ -103,7 +103,7 @@ public static function load($key, ?string $password = null): array
             throw new \UnexpectedValueException('Key should be a string - not a ' . gettype($key));
         }
 
-        $components = ['isPublicKey' => strpos($key, 'PUBLIC') !== false];
+        $components = ['isPublicKey' => str_contains($key, 'PUBLIC')  ];
```
Comment (terrafrost) [inline phpseclib/Crypt/RSA/Formats/Keys/PSS.php:106]: Trailing white space should be removed here

File: phpseclib/Net/SSH2.php [hunk 3]
```diff
@@ -1499,16 +1499,16 @@ private function key_exchange($kexinit_payload_server = false): bool
 
         $exchange_hash_rfc4419 = '';
 
-        if (strpos($this->kex_algorithm, 'curve25519-sha256') === 0 || strpos($this->kex_algorithm, 'ecdh-sha2-nistp') === 0) {
-            $curve = strpos($this->kex_algorithm, 'curve25519-sha256') === 0 ?
+        if (str_starts_with($this->kex_algorithm, 'curve25519-sha256')   || str_starts_with($this->kex_algorithm, 'ecdh-sha2-nistp')) {
+            $curve = str_starts_with($this->kex_algorithm, 'curve25519-sha256')   ?
```
Comment (terrafrost) [inline phpseclib/Net/SSH2.php:1503]: Looks like there are some double spaces here that ought to be reduced to a single space

File: phpseclib/Net/SSH2.php [hunk 4]
```diff
@@ -3263,18 +3263,18 @@ private function get_binary_packet(bool $skip_channel_filter = false)
                     $cmf = ord($payload[0]);
                     $cm = $cmf & 0x0F;
                     if ($cm != 8) { // deflate
-                        user_error("Only CM = 8 ('deflate') is supported ($cm)");
+                        trigger_error("Only CM = 8 ('deflate') is supported ($cm)");
```
Comment (terrafrost) [inline phpseclib/Net/SSH2.php:3266]: These should be turned into Exceptions. This should be updated in the 3.0 branch too. I'll backport your changes to master!

File: phpseclib/Crypt/Blowfish.php [hunk 5]
```diff
@@ -536,11 +536,11 @@ private static function expand0state(array $key, array &$sbox0, array &$sbox1, a
             $p[14] ^ $key[14],
             $p[15] ^ $key[15],
             $p[16] ^ $key[0],
-            $p[17] ^ $key[1]
+            $p[17] ^ $key[1],
         ];
 
         // @codingStandardsIgnoreStart
-        [ $p[0],  $p[1]] = self::encryptBlockHelper(     0,      0, $sbox0, $sbox1, $sbox2, $sbox3, $p);
+        [ $p[0],  $p[1]] = self::encryptBlockHelper(     0, 0, $sbox0, $sbox1, $sbox2, $sbox3, $p);
```
Comment (terrafrost) [inline phpseclib/Crypt/Blowfish.php:543]: I think I liked it better before this change. Before, everything lined up:

```
        // @codingStandardsIgnoreStart
        [ $p[0],  $p[1]] = self::encryptBlockHelper(     0,      0, $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[2],  $p[3]] = self::encryptBlockHelper($p[ 0], $p[ 1], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[4],  $p[5]] = self::encryptBlockHelper($p[ 2], $p[ 3], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[6],  $p[7]] = self::encryptBlockHelper($p[ 4], $p[ 5], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[8],  $p[9]] = self::encryptBlockHelper($p[ 6], $p[ 7], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[10], $p[11]] = self::encryptBlockHelper($p[ 8], $p[ 9], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[12], $p[13]] = self::encryptBlockHelper($p[10], $p[11], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[14], $p[15]] = self::encryptBlockHelper($p[12], $p[13], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[16], $p[17]] = self::encryptBlockHelper($p[14], $p[15], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        // @codingStandardsIgnoreEnd
```

Now it doesn't:

```
        // @codingStandardsIgnoreStart
        [ $p[0],  $p[1]] = self::encryptBlockHelper(     0, 0, $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[2],  $p[3]] = self::encryptBlockHelper($p[ 0], $p[ 1], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[4],  $p[5]] = self::encryptBlockHelper($p[ 2], $p[ 3], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[6],  $p[7]] = self::encryptBlockHelper($p[ 4], $p[ 5], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [ $p[8],  $p[9]] = self::encryptBlockHelper($p[ 6], $p[ 7], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[10], $p[11]] = self::encryptBlockHelper($p[ 8], $p[ 9], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[12], $p[13]] = self::encryptBlockHelper($p[10], $p[11], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[14], $p[15]] = self::encryptBlockHelper($p[12], $p[13], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        [$p[16], $p[17]] = self::encryptBlockHelper($p[14], $p[15], $sbox0, $sbox1, $sbox2, $sbox3, $p);
        // @codingStandardsIgnoreEnd
```

File: phpseclib/Crypt/Blowfish.php [hunk 6]
```diff
@@ -605,11 +605,11 @@ private static function expandstate(array $data, array $key, array &$sbox0, arra
             $p[14] ^ $key[14],
             $p[15] ^ $key[15],
             $p[16] ^ $key[0],
-            $p[17] ^ $key[1]
+            $p[17] ^ $key[1],
         ];
 
         // @codingStandardsIgnoreStart
-        [ $p[0],  $p[1]] = self::encryptBlockHelper($data[ 0]         , $data[ 1]         , $sbox0, $sbox1, $sbox2, $sbox3, $p);
+        [ $p[0],  $p[1]] = self::encryptBlockHelper($data[ 0], $data[ 1], $sbox0, $sbox1, $sbox2, $sbox3, $p);
```
Comment (terrafrost) [inline phpseclib/Crypt/Blowfish.php:612]: Same thing as above.

File: phpseclib/Crypt/Blowfish.php (hunk 5 above)
Comment (jack-worman) [reply phpseclib/Crypt/Blowfish.php]: Using loop so we don't have to use special styling

File: phpseclib/Crypt/Blowfish.php (hunk 6 above)
Comment (jack-worman) [reply phpseclib/Crypt/Blowfish.php]: Using loop so we don't have to use special styling

File: phpseclib/Exception/ExceptionInterface.php [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpseclib3\Exception;
+
+/**
+ * Used to mark exceptions originating from this library.
```
[the remark this answers is not in the document]
Comment (jack-worman) [reply phpseclib/Exception/ExceptionInterface.php:8]: Separate PR

**Pull request merged**